src/app-layer-detect-proto.c

```diff
@@ -1297,7 +1297,7 @@ static int AppLayerProtoDetectPMPrepareMpm(AppLayerProtoDetectPMCtx *ctx)
     int ret = 0;
     MpmCtx *mpm_ctx = &ctx->mpm_ctx;
 
-    if (mpm_table[mpm_ctx->mpm_type].Prepare(mpm_ctx) < 0)
+    if (mpm_table[mpm_ctx->mpm_type].Prepare(mpm_ctx, false) < 0)
```

Comment:
I'm not too excited about this bool everywhere, that isn't really clear. Wonder if it should be a flags field with named flags, like `MPM_PREPARE_FLAG_NONE, MPM_PREPARE_FLAG_USE_CACHE`, etc or similar.